syscalls.c

```diff
@@ -1868,8 +1868,6 @@ void *syscalls_dispatch(int n, char *ustack)
 		proc_threadEnd();
 	}
 
-	threads_setupUserReturn(retval);
```

Comment:
I think I'd prefer changing `syscalls_dispatch` to `void *syscalls_dispatch(int n, cpu_context_t *ctx, char *ustack)` and keeping this call here, that way it would not have to be added in each asm file.

Reply:
Yeah, youre rigth I switched to this approach